lib/Parse/Lexer.cpp

```diff
@@ -932,20 +932,27 @@ void Lexer::lexHexNumber() {
   const char *TokStart = CurPtr-1;
   assert(*TokStart == '0' && "not a hex literal");
 
-  // 0x[0-9a-fA-F][0-9a-fA-F_]*
-  ++CurPtr;
-  if (!isHexDigit(*CurPtr)) {
-    diagnose(CurPtr, diag::lex_expected_digit_in_int_literal);
+  auto expected_hex_digit = [&]() {
+    diagnose(CurPtr, diag::lex_expected_digit_in_int_literal, 3 /*Hex*/);
```

Comment:
Yes, I think so, although it should only need to be available in the bit of this file starting from above this function, and all the uses should be fairly close together.